manager/allocator/networkallocator/networkallocator.go

```diff
@@ -878,3 +890,16 @@ func serviceNetworks(s *api.Service) []*api.NetworkAttachmentConfig {
 	}
 	return s.Spec.Task.Networks
 }
+
+//IsVipOnIngressNetwork check if the vip is in ingress network
+func (na *NetworkAllocator) IsVipOnIngressNetwork(vip *api.Endpoint_VirtualIP) bool {
+	if vip == nil {
+		return false
+	}
+
+	localNet := na.getNetwork(vip.NetworkID)
+	if localNet != nil && localNet.nw != nil {
+		return localNet.nw.Spec.Ingress
```

Comment:
We could define a `networkallocator.IsIngressNetwork(...)`
and change `allocator.IsIngressNetwork(x ...) { return networkallocator.IsIngressNetwork(x) }`